starter-code/src/clue.js

```diff
@@ -92,5 +92,23 @@ const weaponsArray = [
 ];
 
 // ITERATION 2
+function selectRandom(arr){
+ return arr[Math.floor(Math.random() * arr.length)];
 
+}
+function pickMystery(){
+    
+    
+    var x={
+        suspect :selectRandom.suspectsArray,
+        room : selectRandom.roomsArray,
+        weapon : selectRandom.weaponsArray
+    }
+    return x;
+}
 // ITERATION 3
+function revealMystery(obj){
```

Comment:
this also returns undefined because of the prevous issue

Reply:
> good job overall. You had some mistakes that will improve with a little more practice but you're doing very well, Tarek. Keep it up! :)

really thanks ..... these comments cheers me up every time I read them XD